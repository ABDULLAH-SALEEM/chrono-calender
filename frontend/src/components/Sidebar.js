import React from "react";
import {
  Drawer,
  List,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
  IconButton,
  Box,
  Typography,
  useTheme,
  useMediaQuery,
  Divider,
} from "@mui/material";
import MenuIcon from "@mui/icons-material/Menu";
import DashboardIcon from "@mui/icons-material/Dashboard";
import PersonIcon from "@mui/icons-material/Person";
import SettingsIcon from "@mui/icons-material/Settings";
import LogoutIcon from "@mui/icons-material/Logout";
import MailIcon from "@mui/icons-material/Mail";
import { useAuth } from "../hooks/useAuth";

const menuItems = [
  { text: "Dashboard", icon: <DashboardIcon /> },
  { text: "Profile", icon: <PersonIcon /> },
  { text: "Invitations", icon: <MailIcon /> },
  { text: "Settings", icon: <SettingsIcon /> },
];

const SIDEBAR_WIDTH = 200;
const SIDEBAR_COLLAPSED_WIDTH = 68;

export default function Sidebar({
  selected = 0,
  onSelect,
  collapsed,
  setCollapsed,
}) {
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down("sm"));
  const [open, setOpen] = React.useState(false);
  const { logout } = useAuth();

  const blackBg = theme.palette.grey[900];
  const white = theme.palette.common.white;
  const black = theme.palette.common.black;

  // App logo and name
  const appName = "Chrono";
  const appLogo = (
    <Box
      sx={{
        display: "flex",
        alignItems: "center",
        justifyContent: collapsed ? "center" : "flex-start",
        px: collapsed ? 0 : 2,
        py: 2,
        minHeight: 56,
        width: "100%",
      }}
    >
      {!collapsed && (
        <>
          <img
            src="/logo192.png"
            alt="Chrono Logo"
            style={{
              width: 32,
              height: 32,
              objectFit: "contain",
              marginRight: collapsed ? 0 : 8,
              display: "block",
            }}
          />
          <Typography
            variant="h6"
            fontWeight={800}
            sx={{ color: theme.palette.primary.main, letterSpacing: 1, ml: 1 }}
          >
            {appName}
          </Typography>
        </>
      )}
    </Box>
  );

  const drawerContent = (
    <Box
      sx={{
        width: collapsed ? SIDEBAR_COLLAPSED_WIDTH : SIDEBAR_WIDTH,
        height: "100vh",
        display: "flex",
        flexDirection: "column",
        justifyContent: "space-between",
        bgcolor: "background.default",
        borderRight: `1px solid ${theme.palette.divider}`,
        pt: 0,
        pb: 0,
        position: "relative",
        transition: "width 0.2s cubic-bezier(.4,0,.2,1)",
      }}
    >
      {/* App logo/name at the top */}
      {appLogo}
      <Box
        sx={{
          flex: 1,
          mt: 0,
          display: "flex",
          flexDirection: "column",
          justifyContent: "flex-start",
        }}
      >
        <List>
          {menuItems.map((item, idx) => {
            const isSelected = selected === idx;
            return (
              <ListItem
                key={item.text}
                disablePadding
                sx={{ display: "block" }}
              >
                <ListItemButton
                  sx={{
                    borderRadius: 2,
                    bgcolor: isSelected ? blackBg : "transparent",
                    color: isSelected ? white : black,
                    mb: 1,
                    mx: 1,
                    minHeight: 48,
                    justifyContent: collapsed ? "center" : "flex-start",
                    px: collapsed ? 0 : 2,
                    "&:hover": {
                      bgcolor: isSelected
                        ? blackBg
                        : theme.palette.action.hover,
                    },
                    transition: "all 0.2s cubic-bezier(.4,0,.2,1)",
                  }}
                  onClick={() => {
                    onSelect && onSelect(idx);
                    setOpen(false);
                  }}
                >
                  <ListItemIcon
                    sx={{
                      color: isSelected ? white : black,
                      minWidth: 0,
                      mr: collapsed ? 0 : 2,
                      justifyContent: "center",
                    }}
                  >
                    {item.icon}
                  </ListItemIcon>
                  {!collapsed && (
                    <ListItemText
                      primary={
                        <Typography
                          fontWeight={isSelected ? 700 : 500}
                          fontSize={16}
                        >
                          {item.text}
                        </Typography>
                      }
                    />
                  )}
                </ListItemButton>
              </ListItem>
            );
          })}
        </List>
      </Box>
      <Box sx={{ width: "100%", pb: 0 }}>
        <Divider />
        <ListItemButton
          sx={{
            borderRadius: 2,
            color: theme.palette.error.main,
            justifyContent: collapsed ? "center" : "flex-start",
            px: collapsed ? 0 : 2,
            minHeight: 48,
            width: "100%",

            transition: "all 0.2s cubic-bezier(.4,0,.2,1)",
          }}
          onClick={logout}
        >
          <ListItemIcon
            sx={{
              color: theme.palette.error.main,
              minWidth: 0,
              mr: collapsed ? 0 : 2,
              justifyContent: "center",
            }}
          >
            <LogoutIcon />
          </ListItemIcon>
          {!collapsed && (
            <ListItemText
              primary={
                <Typography fontWeight={700} fontSize={16}>
                  Sign Out
                </Typography>
              }
            />
          )}
        </ListItemButton>
      </Box>
    </Box>
  );

  return (
    <>
      {isMobile ? (
        <>
          {!open && (
            <IconButton
              onClick={() => setOpen(true)}
              sx={{ position: "fixed", top: 0, left: 13, zIndex: 1300 }}
              size="small"
              color="primary"
            >
              <MenuIcon />
            </IconButton>
          )}
          <Drawer
            anchor="left"
            open={open}
            onClose={() => setOpen(false)}
            PaperProps={{
              sx: { width: SIDEBAR_WIDTH, bgcolor: "background.default" },
            }}
          >
            {drawerContent}
          </Drawer>
        </>
      ) : (
        <Box
          sx={{
            width: collapsed ? SIDEBAR_COLLAPSED_WIDTH : SIDEBAR_WIDTH,
            height: "100vh",
            position: "fixed",
            left: 0,
            top: 0,
            bgcolor: "background.default",
            borderRight: `1px solid ${theme.palette.divider}`,
            zIndex: 1200,
            display: "flex",
            flexDirection: "column",
            justifyContent: "space-between",
            pt: 0,
            pb: 0,
            transition: "width 0.2s cubic-bezier(.4,0,.2,1)",
          }}
        >
          {drawerContent}
        </Box>
      )}
    </>
  );
}
