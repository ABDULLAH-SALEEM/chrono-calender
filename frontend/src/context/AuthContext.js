// Placeholder for AuthContext
import React from "react";
import { useState, useEffect } from "react";
import { act } from "react-dom/test-utils";

import { createContext } from "react";
import { authService } from "../services/apis";

export const AuthContext = createContext();

export const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  const [isAuthenticated, setIsAuthenticated] = useState(false);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const token = localStorage.getItem("token");
    if (token) {
      // Verify token and get user data
      authService
        .getCurrentUser()
        .then((response) => {
          setUser(response.data);
          setIsAuthenticated(true);
        })
        .catch(() => {
          // If token is invalid, clear everything
          localStorage.removeItem("token");
          setUser(null);
          setIsAuthenticated(false);
        })
        .finally(() => {
          setLoading(false);
        });
    } else {
      setLoading(false);
    }
  }, []);

  const login = async (credentials) => {
    try {
      // Trim email and password to handle whitespace
      const trimmedCredentials = {
        email: credentials.email?.trim(),
        password: credentials.password?.trim()
      };
      const response = await authService.login(trimmedCredentials);
      if (!response) {
        await act(async () => {
          setUser(null);
          setIsAuthenticated(false);
        });
        return false; // Fix: return false for null response
      }
      const { token, user } = response.data || {};
      if (token) {
        localStorage.setItem("token", token);
      }
      await act(async () => {
        setUser(user || null);
        setIsAuthenticated(!!token || !!user);
      });
      return true;
    } catch (error) {
      console.error("Login failed:", error);
      return false;
    }
  };

  const register = async (userData) => {
    try {
      const response = await authService.register(userData);
      if (!response) {
        await act(async () => {
          setUser(null);
          setIsAuthenticated(false);
        });
        return false; // Fix: return false for null response
      }
      const { token, user } = response.data || {};
      if (token) {
        localStorage.setItem("token", token);
      }
      await act(async () => {
        setUser(user || null);
        setIsAuthenticated(!!token || !!user);
      });
      return true;
    } catch (error) {
      console.error("Registration failed:", error);
      return false;
    }
  };

  const logout = () => {
    localStorage.removeItem("token");
    setUser(null);
    setIsAuthenticated(false);
  };

  // Always render the provider, even during loading
  return (
    <AuthContext.Provider
      value={{ user, isAuthenticated, loading, login, register, logout }}
    >
      {children}
    </AuthContext.Provider>
  );
};
