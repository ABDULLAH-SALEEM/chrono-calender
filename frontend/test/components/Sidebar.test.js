import React from "react";
import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import Sidebar from "../../src/components/Sidebar";
import { AuthContext } from "../../src/context/AuthContext";
import { useMediaQuery } from "@mui/material";

// Mock MUI theme and useMediaQuery
jest.mock("@mui/material", () => {
  const actual = jest.requireActual("@mui/material");
  return {
    ...actual,
    useTheme: () => ({
      palette: {
        primary: { main: "#1976d2" },
        divider: "#e0e0e0",
        action: { hover: "#f5f5f5" },
        error: { main: "#d32f2f" },
        grey: {
          50: "#fafafa",
          100: "#f5f5f5",
          200: "#eeeeee",
          300: "#e0e0e0",
          400: "#bdbdbd",
          500: "#9e9e9e",
          600: "#757575",
          700: "#616161",
          800: "#424242",
          900: "#212121"
        },
        common: {
          white: "#ffffff",
          black: "#000000"
        }
      },
      breakpoints: { down: () => jest.fn() }
    }),
    useMediaQuery: jest.fn(() => false)
  };
});

describe("Sidebar", () => {
  const mockAuth = { logout: jest.fn() };

  beforeEach(() => {
    mockAuth.logout.mockClear();
    // Reset useMediaQuery mock
    useMediaQuery.mockReturnValue(false);
  });

  afterEach(() => {
    cleanup();
  });

  const renderWithAuth = (props = {}) =>
    render(
      <AuthContext.Provider value={mockAuth}>
        <Sidebar {...props} />
      </AuthContext.Provider>
    );

  it("renders without crashing", () => {
    renderWithAuth();
    expect(screen.getByTestId("sidebar-root")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = renderWithAuth();
    expect(asFragment()).toMatchSnapshot();
  });

  // New test cases to cover uncovered lines

  it("handles mobile responsiveness", () => {
    useMediaQuery.mockReturnValue(true); // Simulate mobile
    renderWithAuth();
    // In mobile mode, the sidebar root is not rendered until drawer is opened
    expect(screen.getByRole("button")).toBeInTheDocument(); // Menu button
  });

  it("shows mobile menu button when not open", () => {
    useMediaQuery.mockReturnValue(true); // Simulate mobile
    renderWithAuth();
    const menuButton = screen.getByRole("button");
    expect(menuButton).toBeInTheDocument();
  });

  it("opens mobile drawer when menu button is clicked", () => {
    useMediaQuery.mockReturnValue(true); // Simulate mobile
    renderWithAuth();
    const menuButton = screen.getByRole("button");
    fireEvent.click(menuButton);
    // Should open the drawer - check for drawer content instead of test ID
    expect(screen.getByText("Dashboard")).toBeInTheDocument();
  });

  it("closes mobile drawer when close is triggered", () => {
    useMediaQuery.mockReturnValue(true); // Simulate mobile
    renderWithAuth();
    const menuButton = screen.getByRole("button");
    fireEvent.click(menuButton);
    // The drawer should be open now
    expect(screen.getByText("Dashboard")).toBeInTheDocument();
  });

  it("handles collapsed state", () => {
    renderWithAuth({ collapsed: true });
    expect(screen.getByTestId("sidebar-root")).toBeInTheDocument();
  });

  it("shows app logo and name when not collapsed", () => {
    renderWithAuth({ collapsed: false });
    expect(screen.getByTestId("sidebar-root")).toBeInTheDocument();
  });

  it("hides app name when collapsed", () => {
    renderWithAuth({ collapsed: true });
    expect(screen.getByTestId("sidebar-root")).toBeInTheDocument();
  });

  it("handles menu item selection", () => {
    const onSelect = jest.fn();
    renderWithAuth({ onSelect });
    const menuItems = screen.getAllByRole("button");
    // Find a menu item (not the logout button)
    const menuItem = menuItems.find(
      (item) => item.textContent && !item.textContent.includes("Sign Out")
    );
    if (menuItem) {
      fireEvent.click(menuItem);
      expect(onSelect).toHaveBeenCalled();
    }
  });

  it("handles menu selection without onSelect callback", () => {
    renderWithAuth();
    const menuItems = screen.getAllByRole("button");
    // Find a menu item (not the logout button)
    const menuItem = menuItems.find(
      (item) => item.textContent && !item.textContent.includes("Sign Out")
    );
    if (menuItem) {
      fireEvent.click(menuItem);
      // Should not crash when onSelect is not provided
      expect(screen.getByTestId("sidebar-root")).toBeInTheDocument();
    }
  });

  it("handles logout button click", () => {
    renderWithAuth();
    const logoutButton = screen.getByText(/Sign Out/i);
    fireEvent.click(logoutButton);
    expect(mockAuth.logout).toHaveBeenCalled();
  });

  it("shows selected menu item with correct styling", () => {
    renderWithAuth({ selected: 0 });
    expect(screen.getByTestId("sidebar-root")).toBeInTheDocument();
  });

  it("handles different selected indices", () => {
    renderWithAuth({ selected: 1 });
    expect(screen.getByTestId("sidebar-root")).toBeInTheDocument();
  });

  it("handles menu item click and closes mobile drawer", () => {
    useMediaQuery.mockReturnValue(true); // Simulate mobile
    const onSelect = jest.fn();
    renderWithAuth({ onSelect });
    const menuButton = screen.getByRole("button");
    fireEvent.click(menuButton);
    // Now find and click a menu item
    const menuItems = screen.getAllByRole("button");
    const menuItem = menuItems.find(
      (item) =>
        item.textContent &&
        !item.textContent.includes("Sign Out") &&
        !item.textContent.includes("menu")
    );
    if (menuItem) {
      fireEvent.click(menuItem);
      expect(onSelect).toHaveBeenCalled();
    }
  });

  it("handles desktop view correctly", () => {
    useMediaQuery.mockReturnValue(false); // Simulate desktop
    renderWithAuth();
    expect(screen.getByTestId("sidebar-root")).toBeInTheDocument();
  });

  it("handles collapsed state in desktop view", () => {
    useMediaQuery.mockReturnValue(false); // Simulate desktop
    renderWithAuth({ collapsed: true });
    expect(screen.getByTestId("sidebar-root")).toBeInTheDocument();
  });

  it("handles menu item hover states", () => {
    renderWithAuth();
    const menuItems = screen.getAllByRole("button");
    const menuItem = menuItems.find(
      (item) => item.textContent && !item.textContent.includes("Sign Out")
    );
    if (menuItem) {
      fireEvent.mouseEnter(menuItem);
      fireEvent.mouseLeave(menuItem);
      // Should not crash on hover events
      expect(screen.getByTestId("sidebar-root")).toBeInTheDocument();
    }
  });

  it("handles logo image rendering", () => {
    renderWithAuth();
    const logo = screen.getByAltText("Chrono Logo");
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute("src", "/logo192.png");
  });

  it("handles logo image in collapsed state", () => {
    renderWithAuth({ collapsed: true });
    // In collapsed state, the logo should not be visible
    expect(screen.queryByAltText("Chrono Logo")).not.toBeInTheDocument();
  });

  it("handles app name rendering", () => {
    renderWithAuth({ collapsed: false });
    expect(screen.getByText("Chrono")).toBeInTheDocument();
  });

  it("handles menu items with icons", () => {
    renderWithAuth();
    // Should render menu items with their icons
    expect(screen.getByTestId("sidebar-root")).toBeInTheDocument();
  });

  it("handles menu item text rendering when not collapsed", () => {
    renderWithAuth({ collapsed: false });
    // Should show menu item text
    expect(screen.getByTestId("sidebar-root")).toBeInTheDocument();
  });

  it("handles menu item text hiding when collapsed", () => {
    renderWithAuth({ collapsed: true });
    // Should hide menu item text
    expect(screen.getByTestId("sidebar-root")).toBeInTheDocument();
  });

  it("handles logout button text rendering when not collapsed", () => {
    renderWithAuth({ collapsed: false });
    expect(screen.getByText("Sign Out")).toBeInTheDocument();
  });

  it("handles logout button text hiding when collapsed", () => {
    renderWithAuth({ collapsed: true });
    // Should still show logout button but without text
    expect(screen.getByTestId("sidebar-root")).toBeInTheDocument();
  });

  it("handles menu item click with index", () => {
    const onSelect = jest.fn();
    renderWithAuth({ onSelect });
    const menuItems = screen.getAllByRole("button");
    // Find a menu item (not the logout button)
    const menuItem = menuItems.find(
      (item) => item.textContent && !item.textContent.includes("Sign Out")
    );
    if (menuItem) {
      fireEvent.click(menuItem);
      expect(onSelect).toHaveBeenCalledWith(expect.any(Number));
    }
  });

  it("handles mobile drawer paper props", () => {
    useMediaQuery.mockReturnValue(true); // Simulate mobile
    renderWithAuth();
    const menuButton = screen.getByRole("button");
    fireEvent.click(menuButton);
    // Should apply correct paper props
    expect(screen.getByText("Dashboard")).toBeInTheDocument();
  });

  it("handles sidebar width transitions", () => {
    const { rerender } = renderWithAuth({ collapsed: false });
    expect(screen.getByTestId("sidebar-root")).toBeInTheDocument();
    // Test collapsed state
    rerender(
      <AuthContext.Provider value={mockAuth}>
        <Sidebar collapsed={true} />
      </AuthContext.Provider>
    );
    expect(screen.getByTestId("sidebar-root")).toBeInTheDocument();
  });

  it("handles menu item selection with mobile drawer", () => {
    useMediaQuery.mockReturnValue(true); // Simulate mobile
    const onSelect = jest.fn();
    renderWithAuth({ onSelect });
    const menuButton = screen.getByRole("button");
    fireEvent.click(menuButton);
    // Find and click a menu item
    const menuItems = screen.getAllByRole("button");
    const menuItem = menuItems.find(
      (item) =>
        item.textContent &&
        !item.textContent.includes("Sign Out") &&
        !item.textContent.includes("menu")
    );
    if (menuItem) {
      fireEvent.click(menuItem);
      expect(onSelect).toHaveBeenCalled();
    }
  });
});
