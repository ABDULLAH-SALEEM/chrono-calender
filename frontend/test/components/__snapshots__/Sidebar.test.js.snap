// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1cyt1mf"
    data-testid="sidebar-root"
  >
    <div
      class="MuiBox-root css-fycgp9"
    >
      <div
        class="MuiBox-root css-1rux0hy"
      >
        <img
          alt="Chrono Logo"
          src="/logo192.png"
          style="width: 32px; height: 32px; object-fit: contain; margin-right: 8px; display: block;"
        />
        <h6
          class="MuiTypography-root MuiTypography-h6 css-y5n0s1-MuiTypography-root"
        >
          Chrono
        </h6>
      </div>
      <div
        class="MuiBox-root css-1y2kpw7"
      >
        <ul
          class="MuiList-root MuiList-padding css-b7dj7n-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters css-1li41ks-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-pm5ggk-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-w5m6wr-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="DashboardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 13h8V3H3zm0 8h8v-6H3zm10 0h8V11h-8zm0-18v6h8V3z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1u7zyji-MuiTypography-root"
                >
                  Dashboard
                </p>
              </div>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-1li41ks-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-nlckjd-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-17vitqo-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="MailIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 4-8 5-8-5V6l8 5 8-5z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-sh1c8h-MuiTypography-root"
                >
                  Invitations
                </p>
              </div>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-1li41ks-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-nlckjd-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-17vitqo-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="SettingsIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-sh1c8h-MuiTypography-root"
                >
                  Settings
                </p>
              </div>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="MuiBox-root css-1kw89le"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
        />
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-15xc6h4-MuiButtonBase-root-MuiListItemButton-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root css-pkhrhz-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="LogoutIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1u7zyji-MuiTypography-root"
            >
              Sign Out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
